odbc/2.ibmi.md

```diff
@@ -0,0 +1,96 @@
+# IBM i ODBC Setup
+
+This page will help you set up your ODBC environment for accessing Db2 directly from your IBM i (either your local machine or remote IBM i systems). If you want to connect to your IBM i from a [Linux](./2.linux.md) or [Windows](./2.windows.md) machine, please consult those pages.
+
+## Driver Manager Installation
+
+On IBM i, we will be using unixODBC as our driver manager. Like all open-source software, you need to use yum to install these packages, which will require you to be using IBM 7.2 or greater. If you do not yet have yum installed on your system, [please consult the installation guide here.](https://ibm.biz/ibmi-rpms)
+
+To make sure that unixODBC is available (it should be, if you have yum installed), run `yum list available unixODBC*` to get a print out of all packages matching that name:
+
+```bash
+$ yum list available unixODBC*
+Available Packages
+unixODBC.ppc64                          2.3.4-2                     Artifactory 
+unixODBC-devel.ppc64                    2.3.4-2                     Artifactory 
+```
+
+You should see both `unixODBC` and `unixODBC-devel` packages displayed. To develop your applications using ODBC, you will need to download both both of these packages.
+
+```bash
+$ yum install unixODBC unixODBC-devel
+```
+
+You now have unixODBC installed on your system. This serves as your driver manager, allowing the ODBC applications you write to communicate with your drivers and from there Db2. 
+
+## Driver Installation
+
+Now that you have the driver manager installed, you will have to install the ODBC driver that allows your IBM i machine to use unixODBC to talk to Db2. This driver is distributed through [Entitled Systems Support Website](https://www.ibm.com/servers/eserver/ess/index.wss). **Please note, you will need an IBMid and a license for an IBM i system.** It is listed with Access Client Solutions downloads under the name `ODBC driver for IBM i 7.2 or later`.
+
+When the package has been downloaded and unzipped, you can run the rpm with yum the same way you would otherwise, but giving it the location of the file instead of the name of the package:
+
+```bash
+$ yum install <package-location>/ibm-iaccess.rpm
+```
+
+This will install the Db2 ODBC driver onto your IBM i system. It will also create a DSN for your local system called `*LOCAL`. This is discussed below.
+
+## Configuration Files
+
+When you have both the driver manager and the driver installed on your system, you will need to set up a few configuration files that define your drivers and datasources for unixODBC.
+
+### **`odbcinst.ini`**
+
+To define your driver so that it can be referenced in your datasource, you will have to define your driver in a file called `odbcinst.ini`. This file can be found at `/etc/unixODBC/odbcinst.ini`. In this file, you define a name for each of your drivers, and tell unixODBC where it can find the driver. an example `odbcinst.ini` might look like:
```

Comment:
It's `/QOpenSys/etc/odbcinst.ini`, actually. odbc.ini is also at `/QOpenSys/etc/odbc.ini`

Probably should mention for unixODBC, the path can be found by running `odbcinst -j`:

```
-bash-4.4$ odbcinst -j
unixODBC 2.3.4
DRIVERS............: /QOpenSys/etc/odbcinst.ini
SYSTEM DATA SOURCES: /QOpenSys/etc/odbc.ini
FILE DATA SOURCES..: /QOpenSys/etc/ODBCDataSources
USER DATA SOURCES..: /home/kadler/.odbc.ini
SQLULEN Size.......: 8
SQLLEN Size........: 8
SQLSETPOSIROW Size.: 8
```